src/main/cpp/option_processor.cc

```diff
@@ -459,16 +459,17 @@ static bool IsValidEnvName(const char* p) {
 
 #if defined(COMPILER_MSVC)
 static void PreprocessEnvString(string* env_str) {
-  static std::set<string> vars_to_uppercase = {"PATH", "TMP", "TEMP", "TEMPDIR",
-                                               "SYSTEMROOT"};
+  static const string vars_to_uppercase[] = {"PATH", "SYSTEMROOT", "TEMP",
+                                             "TEMPDIR", "TMP"};
 
   int pos = env_str->find_first_of('=');
   if (pos == string::npos) return;
 
   string name = env_str->substr(0, pos);
   // We do not care about locale. All variable names are ASCII.
   std::transform(name.begin(), name.end(), name.begin(), ::toupper);
-  if (vars_to_uppercase.find(name) != vars_to_uppercase.end()) {
+  if (std::find(std::begin(vars_to_uppercase), std::end(vars_to_uppercase),
```

Comment:
`std::set` is pretty heavyweight and much slower as it is a binary tree, even binary search on a small flat container is faster than small `std::set`. `std::set` is more suitable if there are more entries.

The runtime difference per function call might be small, but since `PreprocessEnvString` will be run for each environment variable (and there are at least 40 env vars on clean command prompt and at least 70+ in Visual Studio command prompt), the difference will add up.

> It was more readable before IMO.

That I at least agree,

Reply:
Do you have some benchmark data?

Reply:
I will create and run some benchmark when I am free. Thanks for reviewing my other PRs!